docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,379 @@
+and the frequency of purging are either configured or determined automatically
+based on the average block time of the system.
+
+The Completer 
+=============
+
+The Completer is responsible for making sure Blocks and Batches are complete
+before they are delivered. Blocks are considered formally complete once all of
+their predecessors have been delivered to the ChainController and their batches
+field contains all the Batches specified in the BlockHeader’s batch_ids list.
+The batches field is also expected be in the same order as the batch_ids. Once
+Blocks are formally complete they are delivered to the ChainController for
+validation.
+
+Each batch is considered complete once all of its dependent transactions (from
+earlier blocks or batches) have been processed (either by the ChainController
+or the BlockPublisher respectively).
+
+All Blocks and Batches will have a time out for being completed. After the
+initial request for the missing dependencies is sent, if the response is not
+received in a specified time window they are dropped.
+
+If you have a new block of unknown validity, you must ensure that its
+predecessors have been delivered to the journal. If a predecessor is not
+delivered on request to the journal in a reasonable amount of time, the new
+block cannot be validated.
+
+Consider the case where you have the chain A->B->C :
+
+If C arrives and B is not in the BlockCache, the validator will request B. If
+the request for B times out, the C block is dropped.
+
+If later on D arrives with predecessor C, of chain A->B->C->D, the Completer
+will request C from the network and once C arrives, then will request B again.
+If B arrives this time then the chain will be validated.
+
+The Consensus Interface 
+=======================
+
+In the spirit of configurability, the Journal supports pluggable consensus
+algorithms that may be changed via the Configuration transaction family.  The
+selected Consensus algorithm is set for the chain in the Genesis Block during
+Genesis (described below). The Journal and its Consensus Interface support
+pluggable consensus for probabilistic finality algorithms like Proof of Work,
+as well as algorithms with absolute finality like PBFT.
+
+The Consensus algorithm services to the journal are divided into three distinct
+interfaces that have specific lifetimes and access to information.
+
+1. Consensus.BlockPublisher
+2. Consensus.BlockVerifier
+3. Consensus.ForkResolver
+
+Consensus algorithm implementations in Sawtooth Lake must implement all of the
+consensus interfaces. Each of these objects are provided read-only access to
+the BlockCache and GlobalState.
+
+Consensus.BlockPublisher
+------------------------
+
+The Consensus.BlockPublisher implementation of this interface is used by the
+BlockPublisher to create new candidate blocks to extend the chain.  The block
+publisher is provided access to the a read-only view of global state, a
```

Comment:
Too many articles: get rid of either the "the" or the "a":
The block publisher is provided access to the a read-only view of global state,